callback-heaven.lisp

```diff
@@ -257,7 +257,7 @@ Note that this memory is not further managed!"
             (write-char #\; stream)
             (terpri stream)))
 
-(defun emit-api-function-definitions (ctrans stream)
+(defun emit-api-function-definitions (ctrans stream &key prefix postfix)
```

Comment:
I like the hooks here for flexibility. If things were to become unwieldy we could think about using classes and letting the user of callback-heaven create a subclass implementing the desired wrappers.

Reply:
I'm also OK with keeping it simple for now, but another option would be to allow the caller to specify some set of hooks, something like `pre-body-hook`, `post-body-hook`, etc.  Then call those at appropriate points passing the `stream` and maybe some additional info like the current function name, return type, argument names and return types, etc.